examples/gno.land/p/nt/simpledao/dao.gno

```diff
@@ -0,0 +1,206 @@
+		//
+		// In fact, the proposal should stop accepting
+		// votes as soon as a 2/3+ majority is reached
+		// on either option, but leaving the ability to vote still,
+		// even if a proposal is accepted, or not accepted,
+		// leaves room for "principle" vote decisions to be recorded
+		return errProposalInactive
+	}
+
+	// Check if the proposal executor is expired
+	if prop.executor.IsExpired() {
+		return errProposalExpired
+	}
+
+	// Cast the vote
+	if err = prop.votes.castVote(member, option); err != nil {
+		return ufmt.Errorf("unable to vote on proposal %d, %s", id, err)
+	}
+
+	// Check the votes to see if quorum is reached
+	var (
+		majorityPower = s.membStore.getMajorityPower()
+		yays, nays    = prop.votes.getTally()
+	)
+
+	switch {
+	case yays > majorityPower:
+		prop.status = dao.Accepted
+	case nays > majorityPower:
+		prop.status = dao.NotAccepted
+	default:
+		// Quorum not reached
+	}
+
+	return nil
+}
+
+func (s *SimpleDAO) ExecuteProposal(id uint64) error {
+	var (
+		caller      = getDAOCaller()
+		sentCoins   = std.GetOrigSend() // Get the sent coins, if any
+		canCoverFee = sentCoins.AmountOf("ugnot") >= minExecuteFee.Amount
+	)
+
+	// Check if the non-DAO member can cover the execute fee
+	if !s.membStore.IsMember(caller) && !canCoverFee {
+		return errInsufficientExecuteFunds
+	}
+
+	// Check if the proposal exists
+	propRaw, err := s.propStore.GetProposalByID(id)
+	if err != nil {
+		return ufmt.Errorf("unable to get proposal %d, %s", id, err)
+	}
+
+	prop := propRaw.(*proposal)
+
+	// Check the proposal status
+	if prop.GetStatus() != dao.Accepted {
+		// Proposal is not accepted, cannot be executed
+		return errProposalNotAccepted
+	}
+
+	// Check if the proposal is executed
+	if prop.GetStatus() == dao.Executed {
+		// Proposal is already executed
+		return errProposalExecuted
+	}
+
+	// Check if proposal is expired
+	if prop.executor.IsExpired() {
+		return errProposalExpired
+	}
+
+	// Update the proposal status
+	prop.status = dao.Executed
+
+	// Attempt to execute the proposal
+	if err = prop.executor.Execute(); err != nil {
+		return ufmt.Errorf("error during proposal %d execution, %s", id, err)
+	}
+
+	return nil
+}
+
+func (s *SimpleDAO) GetMembStore() dao.MemberStore {
+	return s.membStore
+}
+
+func (s *SimpleDAO) GetPropStore() dao.PropStore {
+	return s.propStore
+}
+
+// getDAOCaller returns the DAO caller.
+// XXX: This is not a great way to determine the caller, and it is very unsafe.
+// However, the current MsgRun context does not persist escaping the main() scope.
+// Until a better solution is developed, this enables proposals to be made through a package deployment + init()
+func getDAOCaller() std.Address {
+	return std.GetOrigCaller()
+}
+
+// TODO change to r/sys/vars
+const daoPkgPath = "gno.land/r/gov/dao"
+
+// isCallerGOVDAO returns a flag indicating if the
+// current caller context is the active GOVDAO
+func isCallerGOVDAO() bool {
```

Comment:
could be okay if the implem was not in `nt/`

or can stay in `nt/` if this is not govdao, but a parameter which is specific to the DAO.

I've a proposal for you on the `type DAO Interface{}`

Reply:
I've moved it out to `p/demo/simpledao`:

[8f2e3ac](https://github.com/gnolang/gno/pull/2581/commits/8f2e3ac737c295e2db07d03b354a95c89f9df0de)